ast/compile.go

```diff
@@ -1905,35 +1910,35 @@ func (f *equalityFactory) Generate(other *Term) *Expr {
 	return expr
 }
 
-const localVarFmt = "__local%d__"
-
 type localVarGenerator struct {
-	exclude   VarSet
-	generated VarSet
+	exclude VarSet
+	next    int
 }
 
-func newLocalVarGenerator(node interface{}) *localVarGenerator {
+func newLocalVarGeneratorForModuleSet(sorted []string, modules map[string]*Module) *localVarGenerator {
 	exclude := NewVarSet()
-	vis := &VarVisitor{
-		vars: exclude,
+	vis := &VarVisitor{vars: exclude}
+	for _, key := range sorted {
+		Walk(vis, modules[key])
 	}
+	return &localVarGenerator{exclude: exclude, next: 0}
+}
+
+func newLocalVarGenerator(node interface{}) *localVarGenerator {
+	exclude := NewVarSet()
+	vis := &VarVisitor{vars: exclude}
 	Walk(vis, node)
-	return &localVarGenerator{exclude, NewVarSet()}
+	return &localVarGenerator{exclude: exclude, next: 0}
 }
 
 func (l *localVarGenerator) Generate() Var {
-	name := Var("")
-	x := 0
-	for len(name) == 0 || l.generated.Contains(name) || l.exclude.Contains(name) {
-		name = Var(fmt.Sprintf(localVarFmt, x))
-		x++
+	for {
+		result := Var("__local" + strconv.Itoa(l.next) + "__")
+		l.next++
```

Comment:
Just to make sure I'm reasoning correctly on `l.next` and its potential to overflow. I'm assuming at the point we had that many local variables in a single compiler instance there would be other problems, and its pretty unrealistic that a policy would be able to generate that many... right?

Reply:
Right. 2**32 unique variables would certainly be an issue.